ESP8266Interface.cpp

```diff
@@ -46,8 +46,16 @@ int ESP8266Interface::connect(const char *ssid, const char *pass, nsapi_security
 
 int ESP8266Interface::connect()
 {
+    _esp.setTimeout(ESP8266_MISC_TIMEOUT);
+    
+    if(!_esp.check_firmware_version()) {
+        debug("ERROR: Firmware incompatible with this driver.\
```

Comment:
Or actually, have the current version defined as `ESP8266_VERSION`, then use "%d" in the debug statement so we only have to update one number (and maybe use that define for the version check).